lib/bitstyles_phoenix/components/button.ex

```diff
@@ -14,7 +14,13 @@ defmodule BitstylesPhoenix.Button do
   `Phoenix.HTML.Link.button/2`, with the following additional notes:
 
   `opts[:to]` — if there’s a `to` parameter, you’ll get an anchor element, otherwise a button element.
+    
+  `opts[:link_fn]` — Overrides the function used to generate the anchor element, when `opts[:to]` is provided.
+      By default, the anchor element will be generated with `Phoenix.HTML.Link.link/2`. 
+      `link_fn` must be a function of arity 2, accepting a text and opts as argument.
```

Comment:
We should add some text, that this can be used with live views `live_patch` / `live_redirect`